run/telegram2john.py

```diff
@@ -73,44 +87,41 @@ def tdfs_parser(filename):
 
 # Derived partly from localstorage.cpp -> readFile()
 def process_tdfs_file(base):
-    """
-    # read settings
-    directory = os.path.join(base, "tdata")
-    f = os.path.join(directory, "settings0")
-    if os.path.exists(f):
-        settings = os.path.join(directory, "settings0")
+    # detect the path - allows directories and actual map file
+    if os.path.isfile(base) or "map" in base:
+        map_path = base
     else:
-        settings = os.path.join(directory, "settings1")
-    f = BytesIO(tdfs_parser(settings))
+        if base.endswith('"'): # to deal with the trailing quote ex. "...\Telegram Desktop\"
+            base = base[:-1]
+        user_path = "D877F783D5D3EF8C"
```

Comment:
The user can specify the direct path to the "map" file now.

So, I am OK with hardcoding this `user_path` value.